### PR TITLE
feat(source-gcs): updated support level, logging, cat for oauth

### PR DESCRIPTION
**Pull request by darynaishchenko**
## What
ab_internal: sl: 300
release_stage: certified

Updates logging to wrap warning to airbyte logs (so cat don't fail with them).

Passes project to Client. We don't use requests where project needed so we can go with default option. Without provided project it can fail with auth error.

Use pydantic.v1 to avoid deprecation warnings.

## Can this PR be safely reverted and rolled back?

- [ ] YES 💚
- [ ] NO ❌

### CODE REVIEW

**Comment by darynaishchenko** [comment]
[regression tests](https://github.com/airbytehq/airbyte/actions/runs/11558503184/job/32171035000)
failed on control version with error that this pr fixes (auth exception)

**Review by bazarnov — APPROVED** [review]

File: airbyte-integrations/connectors/source-gcs/source_gcs/stream_reader.py [hunk 1]
```diff
@@ -50,7 +53,8 @@ def _initialize_gcs_client(self):
             raise ValueError("Source config is missing; cannot create the GCS client.")
         if self._gcs_client is None:
             credentials = self._get_credentials()
-            self._gcs_client = storage.Client(credentials=credentials)
+            # using default project to avoid getting project from env
```
Comment (artem1205) [inline airbyte-integrations/connectors/source-gcs/source_gcs/stream_reader.py:56]: please specify that we need this for Oauth credentials only.

btw, I think we can use project from service account if its specified

Comment (darynaishchenko) [reply airbyte-integrations/connectors/source-gcs/source_gcs/stream_reader.py]: updated

**Review by artem1205 — APPROVED** [review]
LGTM!

**Pull request merged**